formats/json-io/commonTest/src/kotlinx/serialization/json/io/IoTests.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2017-2024 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.serialization.json.io
+
+import kotlinx.io.*
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.*
+import kotlinx.serialization.json.io.internal.*
+import kotlin.test.*
+
+class IoTests {
+
+    @Serializable
+    data class Simple(val i: Int)
+
+    @Test
+    fun testSurrogate() {
+        val text = "\uD83D\uDE03"
+        val originalChars = text.toCharArray()
+
+        val buffer = Buffer()
+        buffer.writeString(text)
+        val reader = IoSerialReader(buffer)
+
+        val readArray = CharArray(2)
+        assertEquals(1, reader.read(readArray, 0, 1) )
+        assertEquals(1, reader.read(readArray, 1, 1) )
+
+        assertContentEquals(originalChars, readArray)
+    }
+
+
+    @Test
+    fun testEncodingAndDecoding() {
```

Comment:
Theoretically, IO's `encodeToSink/decodeFromSource` should also be covered by `json-tests` module. Isn't this the case?

Reply:
In `json-tests`, methods with a different signature are being tested (with explicit transfer of the serializer)